src/lib/gssapi/mechglue/g_initialize.c

```diff
@@ -467,11 +467,13 @@ load_if_changed(const char *pathname, time_t last, time_t *highest)
 	time_t mtime;
 
 	mtime = check_link_mtime(pathname, &mtime);
+        if (mtime == (time_t)-1 && last == (time_t)-1)
+            mtime = 0;
```

Comment:
What's this check for?  mtime == -1 still indicates an error happened even if last is also -1.

Reply:
It is to ignore the mtime error on the first invocation (when last == -1), because we do not care what mtime is in that case.
After the first time it behaves like the previous code.

Reply:
Are we expecting an lstat() or stat() error, and do we want to go ahead and try to read the file in that case?  I thought we were just expecting an mtime of 0.

Reply:
I think we should try to read the file even if lstat/stat fails, worst case we'll get an error in loadConfigFile() when trying to fopen(), and then we'll just return, best case we still can read the config file even though the stat failed.

Are you concerned about the possibility of a transient lstat()/stat() error that may resolve in the future?

I think a transient error of that kind gives no guarantees a future reload will work either, as a future stat may return an mtime lower or equal than other config files present and therefore cause the code to skip the file entirely.

To support recovery from transient errors we'd really need to keep a cache of config files with individual timestamp, and perhaps, for filesystems that support it, a file hash. But I think thisis not a failure case worth optimizing for at this time.

Reply:
> Are you concerned about the possibility of a transient lstat()/stat() error that may resolve in the future?

No, I wasn't intending to raise that possibility.  I just don't see how this check is necessary.  Do we expect to encounter a filesystem where lstat() or stat() fails but the file can be read?  I thought we were just trying to deal with synthetic mtimes of 0.

Reply:
Should I just ignore the case when mtime == -1 and leave the code to return immediately like it was doing before ?

Reply:
That would be my preference.

Reply:
done